src/AwsS3Adapter.php

```diff
@@ -551,8 +552,16 @@ protected function upload($path, $body, Config $config)
         $options = $this->getOptionsFromConfig($config);
         $acl = isset($options['ACL']) ? $options['ACL'] : 'private';
 
-        if (!isset($options['ContentType']) && is_string($body)) {
-            $options['ContentType'] = Util::guessMimeType($path, $body);
+        if (!isset($options['ContentType'])) {
+            if (is_string($body)) {
+                $options['ContentType'] = Util::guessMimeType($path, $body);
+            }
+            if (empty($options['ContentType'])) {
```

Comment:
If https://github.com/thephpleague/flysystem/pull/617 gets accepted then we can remove this block and remove $body from the method call above.